File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n) {
    
    // 공식이 있으려나.?
    // 연속된 자연수니깐 그냥 브루트포스 알고리즘
    int answer = 1;
    for(int i=1;i<n+1;i++)
    {
        int a=i;
        int b=i+1;
        while(a<=n)
        {
            a+=b;
            if(a==n)
            {
                cout<<i;
                answer++;
                break;
            }
            b++;
        }
    }
    cout<< ":" << answer ;
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 1
    max = 10000

    # 투 포인터 알고리즘사용
    
    for i in range(1,(n+1)//2) :
        sum =0
        for j in range(i,10001,1) :
            sum+=j
            if n<sum : break
            elif n==sum : 
                answer+=1
                break 
    
    
    return answer